Nightly CI is failing integration tests

Looks like nightly CI started failing last week: https://github.com/containerd/rust-extensions/actions with the first failure in https://github.com/containerd/rust-extensions/actions/runs/5273246763

_Originally posted by @jsturtevant in https://github.com/containerd/rust-extensions/issues/139#issuecomment-1601821216_

Interestingly it was passing and is now failing on the same commit: [35a97f1](https://github.com/containerd/rust-extensions/commit/35a97f17d55753bb1ef04c28cd7c3203993932b0) so something else in the dependency chain must have changed.

The tests failing are consistent:

```
2023-06-15T00:17:12.6609993Z --- FAIL: TestContainerAttachProcess (0.16s)
2023-06-15T00:17:14.0144827Z --- FAIL: TestContainerAttach (0.10s)
```

Comparing https://github.com/containerd/rust-extensions/actions/runs/5261500598/jobs/9509676956 (passing) and https://github.com/containerd/rust-extensions/actions/runs/5273246763/jobs/9536435763 (failing)

The rust version is the same

```
info: latest update on 2023-01-10, rust version 1.66.1 (90743e729 2023-01-10)
```

The containerd commit is the same for `refs/tags/v1.6.21`

```
3dce8eb055cbb6872793272b4f20ed16117344f8
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -100,16 +100,22 @@ jobs:
       - name: Checkout extensions
         uses: actions/checkout@v3
 
-      - uses: actions/setup-go@v4
-        with:
-          go-version: '1.20.4'
-
       - name: Checkout containerd
         uses: actions/checkout@v3
         with:
           repository: containerd/containerd
           path: src/github.com/containerd/containerd
           ref: ${{ matrix.containerd }}
+
+      - name: Get Go Version
+        run: |
+          go_version=$(awk -F': ' '/GO_VERSION/ {gsub(/["'\'']/, "", $2); print $2; exit}' .github/workflows/release.yml)
+          echo "GO_VERSION=$go_version" >> $GITHUB_ENV
+        working-directory: src/github.com/containerd/containerd
+
+      - uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
 
       - name: Install containerd
         env:
@@ -122,7 +128,7 @@ jobs:
           sudo -E PATH=$PATH script/setup/install-runc
           sudo -E PATH=$PATH script/setup/install-cni $(grep containernetworking/plugins go.mod | awk '{print $2}')
           # Install containerd
-          make bin/containerd GO_BUILD_FLAGS="-mod=vendor" BUILDTAGS="no_btrfs no_devmapper"
+          sudo -E PATH=$PATH make bin/containerd GO_BUILD_FLAGS="-mod=vendor" BUILDTAGS="no_btrfs no_devmapper"
           sudo -E PATH=$PATH install bin/containerd /usr/local/bin/
         working-directory: src/github.com/containerd/containerd
 
@@ -130,6 +136,11 @@ jobs:
         run: |
           cargo build --release --bin containerd-shim-runc-v2-rs
           sudo install -D ./target/release/containerd-shim-runc-v2-rs /usr/local/bin/
+
+      ## get latest go version for integrations tests so we can skip runnings tests
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.20.4'
 
       - name: Integration
         env:

